dlt/sources/sql_database/schema_types.py

```diff
@@ -0,0 +1,160 @@
+from typing import (
+    Optional,
+    Any,
+    Type,
+    TYPE_CHECKING,
+    Literal,
+    List,
+    Callable,
+    Union,
+)
+from typing_extensions import TypeAlias
+from dlt.common.libs.sql_alchemy import Table, Column, Row, sqltypes, Select, TypeEngine
+
+
+from dlt.common import logger
+from dlt.common.schema.typing import TColumnSchema, TTableSchemaColumns
+
+ReflectionLevel = Literal["minimal", "full", "full_with_precision"]
+
+
+# optionally create generics with any so they can be imported by dlt importer
+if TYPE_CHECKING:
+    SelectAny: TypeAlias = Select[Any]  # type: ignore[type-arg]
+    ColumnAny: TypeAlias = Column[Any]  # type: ignore[type-arg]
+    RowAny: TypeAlias = Row[Any]  # type: ignore[type-arg]
+    TypeEngineAny = TypeEngine[Any]  # type: ignore[type-arg]
+else:
+    SelectAny: TypeAlias = Type[Any]
+    ColumnAny: TypeAlias = Type[Any]
+    RowAny: TypeAlias = Type[Any]
+    TypeEngineAny = Type[Any]
+
+
+TTypeAdapter = Callable[[TypeEngineAny], Optional[Union[TypeEngineAny, Type[TypeEngineAny]]]]
+
+
+def default_table_adapter(table: Table, included_columns: Optional[List[str]]) -> None:
+    """Default table adapter being always called before custom one"""
+    if included_columns is not None:
+        # Delete columns not included in the load
+        for col in list(table._columns):  # type: ignore[attr-defined]
+            if col.name not in included_columns:
+                table._columns.remove(col)  # type: ignore[attr-defined]
+    for col in table._columns:  # type: ignore[attr-defined]
+        sql_t = col.type
+        if hasattr(sqltypes, "Uuid") and isinstance(sql_t, sqltypes.Uuid):
+            # emit uuids as string by default
+            sql_t.as_uuid = False
+
+
+def sqla_col_to_column_schema(
+    sql_col: ColumnAny,
+    reflection_level: ReflectionLevel,
+    type_adapter_callback: Optional[TTypeAdapter] = None,
+) -> Optional[TColumnSchema]:
+    """Infer dlt schema column type from an sqlalchemy type.
+
+    If `add_precision` is set, precision and scale is inferred from that types that support it,
+    such as numeric, varchar, int, bigint. Numeric (decimal) types have always precision added.
+    """
+    col: TColumnSchema = {
+        "name": sql_col.name,
+        "nullable": sql_col.nullable,
+    }
+    if reflection_level == "minimal":
+        # TODO: when we have a complex column, it should not be added to the schema as it will be
```

Comment:
@rudolfix this is the case I described, the column will be there will be no data and now type as this will end up as subtables.